File: src/test/java/net/loxal/soa/restkit/endpoint/AbstractEndpointTest.java
/*
 * Copyright 2014 Alexander Orlov <alexander.orlov@loxal.net>. All rights reserved.
 */

package net.loxal.soa.restkit.endpoint;

import org.junit.Test;

import javax.ws.rs.client.Client;
import javax.ws.rs.client.ClientBuilder;
import javax.ws.rs.client.WebTarget;
import javax.ws.rs.core.MediaType;
import javax.ws.rs.core.Response;
import java.io.IOException;
import java.net.URI;
import java.util.Properties;
import java.util.logging.Logger;

import static org.junit.Assert.assertEquals;

public abstract class AbstractEndpointTest {
    protected static final String NON_EXISTENT = "non-existent";
    static final Logger LOG = Logger.getGlobal();
    private static final String SERVICE_TARGET;
    private static final Client client = ClientBuilder.newClient();
    public static Properties properties;
    protected static String resourcePath = "/";

    public static WebTarget prepareGenericRequest(String path) {
        return prepareTarget(SERVICE_TARGET).path(path);
    }

    public static WebTarget prepareTarget(String target) {
        return client.target(target);
    }

    public static WebTarget prepareTarget(URI target) {
        return client.target(target);
    }

    static {
        ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
        properties = new Properties();
        try {
            properties.load(classLoader.getResourceAsStream("local.properties"));
        } catch (IOException e) {
            LOG.severe(String.format("Impossible to load property file. \n %s", e.getMessage()));
        }
        SERVICE_TARGET = properties.getProperty("deploymentTarget");
    }

    @Test
    public void nonExistentPathNotFound() throws Exception {
        Response response = prepareGenericRequest(resourcePath).path(NON_EXISTENT).request().get();
        LOG.info("resourcePath: " + resourcePath);

        assertEquals(Response.Status.NOT_FOUND.getStatusCode(), response.getStatus());
        assertEquals(MediaType.APPLICATION_JSON_TYPE, response.getMediaType());
    }
}

File: src/test/kotlin/net/loxal/soa/restkit/endpoint/AbstractEndpointTest.kt
/*
 * Copyright 2016 Alexander Orlov <alexander.orlov@loxal.net>. All rights reserved.
 */

package net.loxal.soa.restkit.endpoint

import com.fasterxml.jackson.databind.DeserializationFeature
import com.fasterxml.jackson.databind.ObjectMapper
import net.loxal.soa.restkit.App
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import java.net.URI
import javax.ws.rs.client.ClientBuilder
import javax.ws.rs.client.WebTarget

abstract class AbstractEndpointTest {

    companion object {
        val LOG: Logger = LoggerFactory.getLogger(AbstractEndpointTest::class.java)
        val NON_EXISTENT: String = "non-existent"
        val SERVICE_TARGET: String
        private val client = ClientBuilder.newClient()
        private val mapper = ObjectMapper()
        var resourcePath: String = "/"

        fun prepareGenericRequest(path: String): WebTarget {
            return prepareTarget(SERVICE_TARGET).path(path)
        }

        fun prepareTarget(target: String): WebTarget {
            return client.target(target)
        }

        fun prepareTarget(target: URI): WebTarget {
            return client.target(target)
        }

        init {
            SERVICE_TARGET = App.PROPERTIES.getProperty("deploymentTarget")

            mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
            client.register(mapper)
        }
    }
}